pkg/build/controller/build/build_controller.go

```diff
@@ -425,12 +425,14 @@ func (bc *BuildController) cancelBuild(build *buildapi.Build) (*buildUpdate, err
 func (bc *BuildController) handleNewBuild(build *buildapi.Build, pod *v1.Pod) (*buildUpdate, error) {
 	// If a pod was found, and it was created after the build was created, it
 	// means that the build is active and its status should be updated
+	// QUESTION 3: if we fix question 2 below, can we remove this entire stanza,
+	// and wouldn't that be the right thing to do?
```

Comment:
So if you already know that a pod exists, why not handle it here instead of checking policy, and then trying to create it?

Reply:
Because you know that you didn't successfully complete the work in the current state, therefore you shouldn't advance to the next.  The implementation related to handling and advancing the state should be in one place for legibility and maintainability.  And precisely one thing only should determine whether the work in the given state is complete.

Reply:
> Because you know that you didn't successfully complete the work in the current state, therefore you shouldn't advance to the next.

But the work is to move to the next state :)


> The implementation related to handling and advancing the state should be in one place for legibility and maintainability

This I think I agree with. It does allow simpler code.

Reply:
trying to remember the rules for the `workqueue.RateLimitingInterface` ... can we assume we will not get duplicate events?  I *thought* the answer was we could *not* assume that.  If so, we would want this stanza to handle it.

Even if we can assume no duplicate events, this still seems like safe code / protection for something unexpected, including a potential bug in the lister/queue/cache stuff that lead to duplicate events.